plugins/modules/change_request.py

```diff
@@ -0,0 +1,409 @@
+  - Tadej Borovsak (@tadeboro)
+
+short_description: Manage ServiceNow change request
+
+description:
+  - Create, delete or update a ServiceNow change request.
+  - For more information, refer to the ServiceNow change management documentation at
+    U(https://docs.servicenow.com/bundle/paris-it-service-management/page/product/change-management/concept/c_ITILChangeManagement.html).
+
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+  - servicenow.itsm.sys_id.info
+  - servicenow.itsm.number.info
+
+seealso:
+  - module: servicenow.itsm.change_request_info
+
+options:
+  state:
+    description:
+      - The state of the change request.
+      - If I(state) value is C(assess) or C(authorize) or C(scheduled) or
+        C(implement) or C(review) or C(closed),
+        I(assignment_group) parameter must be filled in.
+      - For more information on state model and transition,
+        refere to the ServiceNow documentation at
+        U(https://docs.servicenow.com/bundle/paris-it-service-management/page/product/change-management/concept/c_ChangeStateModel.html)
+    choices: [ new, assess, authorize, scheduled,
+               implement, review, closed, canceled, absent ]
+    type: str
+  type:
+    description:
+      - Specify what type of change is required.
+    choices: [ standard, normal, emergency ]
+    type: str
+  requested_by:
+    description:
+      - User who requested the change.
+    type: str
+  assignment_group:
+    description:
+      - The group that the change request is assigned to.
+      - Required if I(state) value is C(assess) or C(authorize) or
+        C(scheduled) or C(implement) or C(review) or C(closed).
+    type: str
+  category:
+    description:
+      - The category of the change request.
+    choices: [ hardware, software, service, system_software, aplication_software,
+               network, telecom, documentation, other ]
+    type: str
+  priority:
+    description:
+      - Priority is based on impact and urgency, and it identifies how quickly
+        the service desk should address the task.
+    choices: [ critical, high, moderate, low ]
+    type: str
+  risk:
+    description:
+      - The risk level for the change.
+    choices: [ high, moderate, low ]
+    type: str
+  impact:
+    description:
+      - Impact is a measure of the effect of an incident, problem,
+        or change on business processes.
+    choices: [ high, medium, low ]
+    type: str
+  urgency:
+    description:
+      - The extent to which resolution of an change request can bear delay.
+    choices: [ low, medium, high ]
+    type: str
+  short_description:
+    description:
+      - A summary of the change request.
+    type: str
+  description:
+    description:
+      - A detailed description of the change request.
+    type: str
+  close_code:
+    description:
+      - Provide information on how the change request was resolved.
+      - This parameter is required if I(state) is C(closed).
```

Comment:
This is not entirely true. Users can omit the `close_code` parameter if the close code is already set on the change request prior to the closed state transition. SO maybe we should reword with into something like "The change request must have this parameter set prior to transitioning to the C(closed) state."